--- a/src/main.cpp	2025-10-27 22:59:43.000000000 +0900
+++ b/src/main.cpp	2026-07-08 19:33:18.353913800 +0900
@@ -134,6 +134,7 @@
     std::wstring sound(L"Notification.Default");
     std::wstring buttons;
     Duration duration = Duration::Short;
+    bool persistent = false;
     bool silent = false;
     bool closeNotify = false;
     bool isTextBoxEnabled = false;
@@ -183,6 +184,8 @@
             id = nextArg(it,
                          L"Missing argument to -id.\n"
                          L"Supply argument as -id \"id\"");
+        } else if (arg == L"-persistent") {
+            persistent = true;
         } else if (arg == L"-silent") {
             silent = true;
         } else if (arg == L"-appid") {
@@ -293,6 +296,7 @@
             app.setButtons(buttons);
             app.setTextBoxEnabled(isTextBoxEnabled);
             app.setDuration(duration);
+            app.setPersistent(persistent);
             app.displayToast(title, body, image);
             return app.userAction();
         } else {
--- a/src/snoretoasts.cpp	2025-10-27 22:59:43.000000000 +0900
+++ b/src/snoretoasts.cpp	2026-07-08 19:32:23.284124100 +0900
@@ -83,6 +83,8 @@
 
     Duration m_duration = Duration::Short;
 
+    bool m_persistent = false;
+
     SnoreToastActions::Actions m_action = SnoreToastActions::Actions::Clicked;
 
     ComPtr<IXmlDocument> m_toastXml;
@@ -155,6 +157,9 @@
     ST_RETURN_ON_ERROR(addAttribute(L"launch", rootAttributes.Get(), data));
     ST_RETURN_ON_ERROR(addAttribute(L"duration", rootAttributes.Get(),
                                     d->m_duration == Duration::Short ? L"short" : L"long"));
+    if (d->m_persistent) {
+        ST_RETURN_ON_ERROR(addAttribute(L"scenario", rootAttributes.Get(), L"reminder"));
+    }
     // Adding buttons
     if (!d->m_buttons.empty()) {
         setButtons(root);
@@ -512,6 +517,11 @@
     d->m_duration = duration;
 }
 
+void SnoreToasts::setPersistent(bool persistent)
+{
+    d->m_persistent = persistent;
+}
+
 Duration SnoreToasts::duration() const
 {
     return d->m_duration;
--- a/src/snoretoasts.h	2025-10-27 22:59:43.000000000 +0900
+++ b/src/snoretoasts.h	2026-07-08 19:32:46.440948800 +0900
@@ -82,6 +82,8 @@
     Duration duration() const;
     void setDuration(Duration duration);
 
+    void setPersistent(bool persistent);
+
     std::wstring formatAction(const SnoreToastActions::Actions &action,
                               const std::vector<std::pair<std::wstring_view, std::wstring_view>>
                                       &extraData = {}) const;
--- a/src/toasteventhandler.cpp	2025-10-27 22:59:43.000000000 +0900
+++ b/src/toasteventhandler.cpp	2026-07-08 19:35:53.904553000 +0900
@@ -79,8 +79,12 @@
             std::wcout << dataMap.at(L"button") << std::endl;
             m_userAction = SnoreToastActions::Actions::ButtonClicked;
         }
-        if (m_toast.useFalbackMode() && !m_toast.pipeName().empty()) {
-            Utils::writePipe(m_toast.pipeName(), m_toast.formatAction(m_userAction));
+        if (!m_toast.pipeName().empty()) {
+            if (m_userAction == SnoreToastActions::Actions::ButtonClicked) {
+                Utils::writePipe(m_toast.pipeName(), data);
+            } else {
+                Utils::writePipe(m_toast.pipeName(), m_toast.formatAction(m_userAction));
+            }
         }
     }
     SetEvent(m_event);
